environment/peercontainer.go

```diff
@@ -162,14 +163,34 @@ func (e *PeerContainer) Download(path string) (string, error) {
 
 // Upload uploads the path from outside the environment into it
 func (e *PeerContainer) Upload(outpath string) (envpath string, err error) {
-	envpath = "/var/tmp/" + uniuri.NewLen(5) + "/" + filepath.Base(outpath)
+	outDir := filepath.Dir(outpath)
+	envDir := "/var/tmp/"
+	tmpImg := "tmp:" + uniuri.NewLen(5)
 	cengine := container.GetContainerEngine()
-	err = cengine.CopyDirsIntoContainer(e.CID, map[string]string{outpath: envpath})
+	err = cengine.CopyDirsIntoImage(e.ImageName, tmpImg, map[string]string{outDir: envDir})
```

Comment:
Upload is always supposed to be to a running container. For a new image, we use NewPeerContainer, which does copy to image.